main.java

```diff
@@ -0,0 +1,175 @@
+                processShipmentUpdates(logger); 
+                return "SUCCESS"; 
+            } catch (final Exception ex) { 
+                logger.log(String.format("Failed to process shipment Updates in %s due to %s", scheduledEvent.getAccount(), ex.getMessage())); 
+                throw new RuntimeException(ex); 
+            } 
+        } 
+     
+     
+        private void processShipmentUpdates(final LambdaLogger logger) throws InterruptedException { 
+     
+            final List<String> bucketsToProcess = Constants.BUCKETS_TO_PROCESS; 
+            final ConcurrentHashMap<String, Pair<Long, String>> latestStatusForTrackingNumber = new ConcurrentHashMap<String, Pair<Long, String>>(); 
+            final ConcurrentHashMap<String, List<KeyVersion>> filesToDelete = new ConcurrentHashMap<String, List<DeleteObjectsRequest.KeyVersion>>(); 
+            bucketsToProcess.parallelStream().forEach(bucketName -> { 
+                final List<KeyVersion> filesProcessed = processEventsInBucket(bucketName, logger, latestStatusForTrackingNumber); 
+                filesToDelete.put(bucketName, filesProcessed); 
+        }); 
+             
+            final AmazonS3 s3Client = EventHandler.getS3Client(); 
+            //Create a new file in the Constants.SUMMARY_BUCKET 
+            logger.log("Map of statuses -> " + latestStatusForTrackingNumber); 
+            String summaryUpdateName = Long.toString(System.currentTimeMillis()); 
+             
+            EventHandler.getS3Client().putObject(Constants.SUMMARY_BUCKET, summaryUpdateName, latestStatusForTrackingNumber.toString()); 
+             
+            long expirationTime = System.currentTimeMillis() + Duration.ofMinutes(1).toMillis(); 
+            while(System.currentTimeMillis() < expirationTime) { 
+                if (s3Client.doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) { 
+                    break; 
+                } 
+                logger.log("waiting for file to be created " + summaryUpdateName); 
+                Thread.sleep(1000); 
+            } 
+             
+            // Before we delete the shipment updates make sure the summary update file exists 
+            if (EventHandler.getS3Client().doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) { 
+                deleteProcessedFiles(filesToDelete); 
+                logger.log("All updates successfully processed"); 
+            } else { 
+                throw new RuntimeException("Failed to write sumary status, will be retried in 15 minutes"); 
+            } 
+             
+        } 
+     
+        private List<KeyVersion> processEventsInBucket(String bucketName, LambdaLogger logger, ConcurrentHashMap<String, Pair<Long, String>> latestStatusForTrackingNumber) { 
+     
+            final AmazonS3 s3Client = EventHandler.getS3Client(); 
+            logger.log("Processing Bucket: " + bucketName); 
+     
+            ObjectListing files = s3Client.listObjects(bucketName); 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code uses an outdated API. [ListObjectsV2 is the revised List Objects API, and we recommend you use this revised API for new application developments](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html).